packages/jest-runtime/src/index.ts

```diff
@@ -982,31 +982,17 @@ class Runtime {
       const vmContext = this._environment.getVmContext();
 
       if (vmContext) {
-        if (typeof compileFunction === 'function') {
```

Comment:
I've intentionally kept the check on line 981 so people can still use `runScript` if they really want to